drivers/scaleway/scaleway.go

```diff
@@ -0,0 +1,376 @@
+// Copyright (C) 2015  Nicolas Lamirault <nicolas.lamirault@gmail.com>
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scaleway
+
+import (
+	//"errors"
+	"fmt"
+	//"os"
+	// "os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/docker/machine/drivers"
+	"github.com/docker/machine/log"
+	"github.com/docker/machine/ssh"
+	"github.com/docker/machine/state"
+
+	"github.com/codegangsta/cli"
+	"github.com/nlamirault/go-scaleway/api"
+)
+
+const (
+	driverName      = "scaleway"
+	dockerConfigDir = "/etc/docker"
+)
+
+type Driver struct {
+	Id             string
+	MachineName    string
+	SSHUser        string
+	SSHPort        int
+	UserId         string
+	Token          string
+	Organization   string
+	Image          string
+	Volumes        string
+	IPAddress      string
+	CaCertPath     string
+	PrivateKeyPath string
+	storePath      string
+}
+
+func init() {
+	drivers.Register(driverName, &drivers.RegisteredDriver{
+		New:            NewDriver,
+		GetCreateFlags: GetCreateFlags,
+	})
+}
+
+func GetCreateFlags() []cli.Flag {
+	return []cli.Flag{
+		cli.StringFlag{
+			Name:   "scaleway-userid",
+			Usage:  "Scaleway UserID",
+			EnvVar: "SCALEWAY_USERID",
+		},
+		cli.StringFlag{
```

Comment:
You should remove this optional argument which will always be unique for a token

Use this always: https://github.com/nlamirault/go-scaleway/blob/master/api/client.go#L76